core/model/work.py

```diff
@@ -1428,6 +1432,282 @@ def assign_appeals(self, character, language, setting, story, cutoff=0.20):
     # that Elasticsearch can parse as a date.
     ELASTICSEARCH_TIME_FORMAT = 'YYYY-MM-DD"T"HH24:MI:SS"."MS'
 
+    @classmethod
```

Comment:
Was this code translated from somewhere?

Reply:
Yessir, to_search_documents() was an in-SQL query that did A LOT within the SQL cluster (query + aggregate + jsonify)
We've moved this to an "in-app" version, once approved I will rename THIS method to to_search_documents()
Methid is below in the same file 
https://github.com/ThePalaceProject/circulation/blob/532b7c79c9f1698b2e002a4daa8603c292334517/core/model/work.py#L1431

Reply:
Got it, thanks! I wanted to give this a quick read through to see if there were any obvious errors in translation. I'm guessing probably not given that the tests pass.

Reply:
Looks correct. :+1: